api_test.go

```diff
@@ -211,3 +211,24 @@ func TestHttpApi(t *testing.T) {
 
 	tests.TestApi(newNodeProvider(ctx))(t)
 }
+
+func TestDirectAPI(t *testing.T) {
+	type args struct {
+		url, header, value string
+	}
+	tests := []struct {
+		name string
+		args args
+	}{
+		{"Success", args{"http://127.0.0.1:5001/foo/bar/api/v0", "Authorization", "Bearer token"}},
```

Comment:
We don't run daemon in CI, so this won't work. Preferably we'd spawn test (go)http server here and check the headers

Reply:
Ah yes good idea, even if daemon was run in CI your suggestion is probably a lot more lightweight to test.

Reply:
While writing a test for the http headers, and the `NewURLApiWithClient` function, I came across this error being returned:
```
mime: no media type
```

I've traced it to this particular line of code https://github.com/ipfs/go-ipfs-http-client/blob/2b48dd36ad170149bd4c62a4194587bf7c704f7e/response.go#L113


The test in question causing this is https://github.com/ipfs/go-ipfs-http-client/blob/2b48dd36ad170149bd4c62a4194587bf7c704f7e/api_test.go#L217 however I was also able to replicate it using the following test

```Golang
func Test_Headers(t *testing.T) {
	var (
		headerToTest        = "Test-Header"
		expectedHeaderValue = "thisisaheadertest"
	)
	ts := httptest.NewServer(
		http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
			val := r.Header.Get(headerToTest)
			if val == expectedHeaderValue {
				w.WriteHeader(400)
				return
			}
			w.WriteHeader(200)
		}),
	)
	addr, err := maAddrFromHTTP(ts.URL)
	if err != nil {
		t.Fatal(err)
	}
	api, err := NewApiWithClient(addr, &http.Client{
		Transport: &http.Transport{
			Proxy:             http.ProxyFromEnvironment,
			DisableKeepAlives: true,
		},
	})
	if err != nil {
		t.Fatal(err)
	}
	if _, err := api.Pin().Ls(context.Background()); err != nil {
		t.Fatal(err)
	}
}

func maAddrFromHTTP(url string) (ma.Multiaddr, error) {
	parts := strings.Split(url, "//")
	addrParts := strings.Split(parts[1], ":")
	maAddr := fmt.Sprintf("/ip4/%v/tcp/%v", addrParts[0], addrParts[1])
	return ma.NewMultiaddr(maAddr)
}
```
I'm not sure if this is due to the way I'm writing the test, as previously written tests aren't causing this error, or if this is something to do with how `Send` handles header parsing